pipelines/templates/build-golang-template.yaml

```diff
@@ -0,0 +1,54 @@
+# This template builds a golang container and pushes it to a container registry, using presidio Makefile.
+# It is used in CI to build the base container as well as all golang based services.
+# parameters:
+#   registry_service_connection: service connection to docker registry.
+#   registry_name: full name of container registry (i.e. presidio.azurecr.io)
+#   make_command_to_run: the phony command in Makefile to execute.
+#   image_name: image name to build.
+#   presidio_label: image label.
+#   presidio_branch_label: final label the container will have (latest or branch-name)
+#   deps_label: label of deps container used when building this container.
+
+parameters:
+  registry_service_connection: '' 
+  registry_name: '' 
+  make_command_to_run: ''
+  image_name: ''
+  presidio_label: ''
+  presidio_branch_label: ''
+  deps_label: ''
+steps:
+- bash: |
+    mkdir -p '$(GOBIN)'
+    mkdir -p '$(GOPATH)/pkg'
+    rm -rf '$(MODULEPATH)'
+    mkdir -p '$(MODULEPATH)'
+    shopt -s extglob
+    mv !(gopath) '$(MODULEPATH)' --force
+    echo '##vso[task.prependpath]$(GOBIN)'
+    echo '##vso[task.prependpath]$(GOROOT)/bin'
```

Comment:
If I understand correctly this script setup go on the agent machine? in a bit of manual intervention steps?

Can you name this step / doc. 
e.g. 'shopt -s' what is it for?

Reply:
this is the ugly part of running go<1.11 on an azure pipelines agent as documented here (notice the go-older tab)

https://docs.microsoft.com/en-us/azure/devops/pipelines/ecosystems/go?view=azure-devops&tabs=go-older

notice this step is only done now when running the integ tests (this used to run on the build as well, but there is no need for it there since we're only doing docker builds now).

the reason we need this to run the integ tests is because they themselves to not run from a containers (based from the golang-base). i've already opened a bug for this here: https://dev.azure.com/csedevil/Presidio-internal/_workitems/edit/1804

added better name

Reply:
you're right, this is not required here and is removed. see my comment to why we need this in the integration tests part:

this is the ugly part of running go<1.11 on an azure pipelines agent as documented here (notice the go-older tab)

https://docs.microsoft.com/en-us/azure/devops/pipelines/ecosystems/go?view=azure-devops&tabs=go-older

the reason we need this to run the integ tests is because they themselves to not run from a containers (based from the golang-base). i've already opened a bug for this here: https://dev.azure.com/csedevil/Presidio-internal/_workitems/edit/1804